Tests/scripts/slack_notifier.py

```diff
@@ -111,56 +114,106 @@ def options_handler():
+        "title": f'{entity_title}',
+        "value": '\n'.join(entities),
+        "short": False
+    }]
 
-def get_entities_fields(entity_title, report_file_name=''):
-    if 'lint' in report_file_name:  # lint case
-        failed_entities = get_failing_unit_tests_file_data()
-    else:
-        failed_entities = get_failed_steps_list()
-    entity_fields = []
-    if failed_entities:
-        entity_fields.append({
-            "title": f'{entity_title} - ({len(failed_entities)})',
-            "value": '\n'.join(failed_entities),
-            "short": False
-        })
-    return entity_fields
 
+def get_failed_unit_tests_attachment(build_url: str, is_sdk_build: bool = False) -> List[Dict]:
+    """
+    Returns the failed unit tests attachment to be reported in Slack.
+    Args:
+        build_url (str): Build URL of the given nightly.
+        is_sdk_build (bool): Whether build is SDK nightly or content nightly.
 
-def get_attachments_for_unit_test(build_url, is_sdk_build=False):
-    unittests_fields = get_entities_fields(entity_title="Failed Unittests", report_file_name="failed_lint_report")
-    color = 'good' if not unittests_fields else 'danger'
-    if not unittests_fields:
-        title = 'Content Nightly Unit Tests - Success' if not is_sdk_build else 'SDK Nightly Unit Tests - Success'
+    Returns:
+        (List[Dict]) Dict wrapped inside a list containing failed unit tests attachment.
+    """
+    if artifact_data := get_artifact_data('failed_lint_report.txt'):
+        artifact_data = artifact_data.split('\n')
+        unittests_fields: Optional[List[Dict]] = get_entities_fields('Failed Unittests', artifact_data)
     else:
-        title = 'Content Nightly Unit Tests - Failure' if not is_sdk_build else 'SDK Nightly Unit Tests - Failure'
-    content_team_attachment = [{
+        unittests_fields = None
+    color: str = 'good' if not unittests_fields else 'danger'
+    build_type: str = 'SDK' if is_sdk_build else 'Content'
+    status = 'Success' if not unittests_fields else 'Failure'
+    title: str = f'{build_type} Nightly Unit Tests - {status}'
+    return [{
         'fallback': title,
         'color': color,
         'title': title,
         'title_link': build_url,
         'fields': unittests_fields
     }]
-    return content_team_attachment
+
+
+def get_coverage_attachment(build_url: str) -> Dict:
+    """
+    Returns content coverage report attachment.
+    Args:
+        build_url (str): Build URL of the nightly.
+
+    Returns:
+        (Dict): Attachment of the coverage.
+    """
+    xml_coverage_data: str = get_artifact_data('coverage_report/coverage.xml')
+    coverage_dict_data: OrderedDict = xmltodict.parse(xml_coverage_data)
+    # TODO handle case if no exists
+    coverage_percent: float = float(coverage_dict_data.get('coverage', dict()).get('@line-rate')) * 100.0
+    return {
+        'fallback': f'Coverage Report Content: {coverage_percent:.2f}% Total Coverage',
+        'color': 'good',
```

Comment:
let's set different colors for different ranges, e.g. 0%-49% danger, 50%-59% warning, 60%-100% good.